git/url_test.go

```diff
@@ -196,13 +196,22 @@ func TestParseURL(t *testing.T) {
 				Path:   "",
 			},
 		},
+		{
+			name:    "fails to parse",
+			url:     "ssh://git@[/tmp/git-repo",
+			wantErr: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			u, err := ParseURL(tt.url)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("got error: %v", err)
+			if tt.wantErr {
+				if err == nil {
+					t.Fatalf("expected error, got nil")
+				}
+				return
 			}
+
 			if u.Scheme != tt.want.Scheme {
 				t.Errorf("expected scheme %q, got %q", tt.want.Scheme, u.Scheme)
 			}
```

Comment:
Just a suggestion; we can use the `assert` package to simplify the upcoming asserts:

```go
if tt.wantErr {
	assert.NotNil(t, err)
	return
}

assert.Equal(t, u.Scheme, tt.want.Scheme)
assert.Equal(t, u.User.Username(), tt.want.User)
assert.Equal(t, u.Host, tt.want.Host)
assert.Equal(t, u.Path, tt.want.Path)
```

Reply:
Yeh I suspect this code is so old it was before the team started using testify. Have changed, thanks.